conifer/backends/fpu/writer.py

```diff
@@ -90,6 +90,7 @@ def unpack(data):
   def from_flat_tree_dictionary(tree, iclass):
     n_nodes = len(tree['feature'])
     nodes = []
+    assert [(sum(weights)) for weights in tree['weight']][0] == 1, f'Oblique splits are not supported by the FPU backend, please use the hls backend'
```

Comment:
For robustness and maintainability, I think this logic could be moved up to the model base class wrapped as a function called `isOblique()` or similar.